transport/grpc/handler.go

```diff
@@ -269,7 +270,9 @@ func handlerErrorToGRPCError(err error, responseWriter *responseWriter) error {
 	message := yarpcStatus.Message()
 	// if the yarpc error has a name, set the header
 	if name != "" {
-		responseWriter.AddSystemHeader(ErrorNameHeader, name)
+		if responseWriter != nil {
+			responseWriter.AddSystemHeader(ErrorNameHeader, name)
+		}
```

Comment:
You should add a comment on the method doc that it is also used from the stream interfaces (that in this case responseWriter should be given as nil)